packages/app/src/app/auth/change-my-password/change-my-password.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, EventEmitter, inject, Output, signal } from '@angular/core';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCard } from '@angular/material/card';
+import { MatError, MatFormFieldModule, MatLabel } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { ChangePasswordInput } from '@reefguide/types';
+import { environment } from '../../../environments/environment';
+import { AuthService } from '../auth.service';
+
+@Component({
+  selector: 'change-my-password',
+  imports: [
+    MatCard,
+    MatFormFieldModule,
+    ReactiveFormsModule,
+    MatInputModule,
+    MatLabel,
+    MatError,
+    MatIconModule,
+    MatButtonModule
+  ],
+  templateUrl: './change-my-password.component.html',
+  styleUrl: './change-my-password.component.scss'
+})
+export class ChangeMyPasswordComponent {
+  private readonly authService = inject(AuthService);
+  private readonly snackBar = inject(MatSnackBar);
+
+  // Triggered when successfully changed
+  @Output() onSuccess = new EventEmitter<void>();
+
+  readonly isSubmitting = signal(false);
+  hideCurrentPassword = true;
+  hideNewPassword = true;
+  hideConfirmPassword = true;
+
+  public changePasswordForm = new FormGroup<{
+    currentPassword: FormControl<string>;
+    newPassword: FormControl<string>;
+    confirmPassword: FormControl<string>;
+  }>({
+    currentPassword: new FormControl('', {
+      nonNullable: true,
+      validators: [Validators.required]
+    }),
+    newPassword: new FormControl('', {
+      nonNullable: true,
+      validators: [Validators.required, Validators.minLength(8)]
+    }),
+    confirmPassword: new FormControl('', {
+      nonNullable: true,
+      validators: [Validators.required, Validators.minLength(8)]
+    })
+  });
+
+  get currentPassword() {
+    return this.changePasswordForm.get('currentPassword')!;
+  }
+
+  get newPassword() {
+    return this.changePasswordForm.get('newPassword')!;
+  }
+
+  get confirmPassword() {
+    return this.changePasswordForm.get('confirmPassword')!;
+  }
+
+  // Change password
+  async changePassword(payload: ChangePasswordInput): Promise<{ message: string }> {
+    // NOTE: this is a separate fetch request for now since
+    // HTTPClient has the default behaviour due to the AuthInterceptor of logging
+    // out if it 401s- for the web-api change password route this is not suitable
+    // as a user may accidentally enter their current password in incorrectly
+    return fetch(`${environment.webApiUrl}/auth/change-password`, {
```

Comment:
fine for now, I think I should isolate a separate `HttClient` instance just for the web-api, which would fix this